src/stories/common/ColonySwitcher.stories.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useState } from 'react';
+import type { Meta, StoryObj } from '@storybook/react';
+import clsx from 'clsx';
+import { usePopperTooltip } from 'react-popper-tooltip';
+import { BrowserRouter as Router } from 'react-router-dom';
+import ColonySwitcher, { ColoniesDropdown, ColonyAvatarWrapper } from '~common/Extensions/ColonySwitcher';
+import { watchlistMock } from '~common/Extensions/ColonySwitcher/consts';
+import { useSelectedColony } from '~common/Extensions/ColonySwitcher/hooks';
+import { useAppContext, useDetectClickOutside, useMobile } from '~hooks';
+import Icon from '~shared/Icon';
+import ColonyDropdownMobile from '~common/Extensions/ColonySwitcher/ColonyDropdownMobile';
+
+const meta: Meta<typeof ColonySwitcher> = {
+  title: 'Common/Colony Switcher',
+  component: ColonySwitcher,
+};
+
+export default meta;
+type Story = StoryObj<typeof ColonySwitcher>;
+
+const ColonySwitcherWithHooks = () => {
+  const { userLoading } = useAppContext();
+  const [isOpen, setIsOpen] = useState<boolean>();
+
+  const { colonyToDisplay, colonyToDisplayAddress } = useSelectedColony(watchlistMock);
+  const isMobile = useMobile();
+  const popperTooltipOffset = !isMobile ? [120, 8] : [0, 8];
+
+  const sortByDate = (firstWatchEntry, secondWatchEntry) => {
+    const firstWatchTime = new Date(firstWatchEntry?.createdAt || 1).getTime();
+    const secondWatchTime = new Date(secondWatchEntry?.createdAt || 1).getTime();
+    return firstWatchTime - secondWatchTime;
+  };
+
+  const ref = useDetectClickOutside({ onTriggered: () => setIsOpen(false) });
+  const { getTooltipProps, setTooltipRef, setTriggerRef } = usePopperTooltip(
+    {
+      delayShow: 200,
+      placement: 'bottom',
+      trigger: 'click',
+      visible: isOpen,
+      interactive: true,
+    },
+    {
+      modifiers: [
+        {
+          name: 'offset',
+          options: {
+            offset: popperTooltipOffset,
+          },
+        },
+      ],
+    },
+  );
+
+  return (
+    <Router>
+      <div className="w-[26.75rem] md:64rem">
```

Comment:
Why did you put the whole component here?
Is it not enough to return just a `<ColonySwitcher/>` here?

Reply:
no it's won't be enough, API not works in storybook so, it's the only way to show list of colonies in dropdown